data_structures/linked_lists/circular_linked_list.rb

```diff
@@ -0,0 +1,107 @@
+# Define a node for the list
+class Node
+  attr_accessor :value, :next
+
+  def initialize(value)
+    @value = value
+    @next = nil
+  end
+end
+
+# Class for circular linked list (last node points to the head node)
+class CircularList
+  attr_reader :head
+
+  def initialize
+    @head = nil
+  end
+
+  def insert_tail(value)
+    newNode = Node.new(value)
+    if @head.nil?
+      @head = newNode
+      @head.next = @head
+    else
+      tempNode = @head
+      tempNode = tempNode.next while tempNode.next != @head
+      newNode.next = @head
+      tempNode.next = newNode
+    end
+  end
+
+  def insert_head(value)
+    newNode = Node.new(value)
+    if @head.nil?
+      @head = newNode
+      @head.next = head
+    else
+      tempNode = @head
+      tempNode = tempNode.next while tempNode.next != @head
+      newNode.next = @head
+      tempNode.next = newNode
+      @head = newNode
+    end
+  end
+
+  def print_list
+    print '['
+    unless @head.nil?
+      printNode = @head
+      while printNode.next != @head
+        print printNode.value.to_s
+        print ', '
+        printNode = printNode.next
+      end
+      print printNode.value
+    end
+    print ']'
+
+    puts(STDOUT.flush)
+  end
+
+  def delete_tail
```

Comment:
I think this line probably should be 

```suggestion
  def delete_head
```

Seems to be accidentally renamed at 906ccac28d03b358e8fa86e6ab3db7894967ea59 :eyes:

Reply:
oh crap